WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt

```diff
@@ -85,27 +84,35 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
         }
 
         sealed class PostCard(
-            open val postCardType: PostCardType
-        ) : Card(if (postCardType == CREATE_FIRST) POST_CARD_CREATE_FIRST else POST_CARD_DRAFT_OR_SCHEDULED) {
-            data class PostCardCreateFirst(
-                override val postCardType: PostCardType,
+            override val type: Type,
+            open val footerLink: FooterLink
+        ) : Card(type) {
+            data class PostCardWithoutPostItems(
```

Comment:
1. Praise (❤️): Kudos on refactoring the `PostCard` the way you did, having the with/without items make the code so much more readable.
2. Suggestion (💡): Consider renaming to `PostCardWithout/WithItems`, thus dropping the extra `Post` from the `PostItems`. I understand why you choose to name it explicitly as `PostItems` and not just `Items`, this is because of the `PostItem` inner data class naming. However, I feel that by removing the duplicate `Post` from the naming here will make the name shorter and will make it explicitly generic. With generic I mean that `PostItems` would be the best alternative if there was multiple other inner data classes, like `PostItem`, `SomeItem`, `OtherItem`, etc. But, since there is, and possible will be, only one inner data class, the `PostItem`, then using `With/WithoutItem` feels enough and even more appropriate as it lets the reader know that their is only one type of items. Wdyt?

PS: If you agree, please do the same everywhere, like:
- `POST_CARD_WITHOUT/WITH_ITEMS` enums
- `PostCardWithout/WithItemsViewHolder` classes
- `my_site_post_card_with\without_items` layouts

Reply:
I actually thought about naming classes on lines of `PostCardWithItems/PostCardWithoutItems` but, to me, the item here kind of clashed with the item in the parent class `MySiteCardAndItem`. I also thought about naming them as `PostCardWithPosts/PostCardWithoutPosts` but then went with the full explicit naming to be very clear of what to expect from it 😅.

I can go ahead and drop the extra "post" as per suggestion if you feel strongly about it.

Reply:
👋 @ashiagr !

Thanks so much for the explanation, I didn't consider the `MySiteCardAndItem` naming and how having the `PostCardWithItems` might clash with that parent class. You are right, let's keep things as is for now and maybe revisit that when it becomes more clear, for example, when and if we end up splitting the `MySiteCardAndItem` into `MySiteCard` and `MySiteItem`... 🤞 😅